Thanks. I've changed the integration test to use `os.linesep`; however, I've had a hard time actually testing this on Windows. My initial attempts to run the test on Windows resulted in an odd exception. After a few rounds of debugging, I came to the conclusion that file.line is already broken on Windows for reasons unrelated to my change. 

The following test reproduces the issue. I first ran it under the 2016.11 and 2017.7.2 branches in dev mode on Windows 10. Next I ran it using the Salt-Minion-2017.7.2-AMD64-Setup.exe Python 2 and 3 installers from https://docs.saltstack.com/en/latest/topics/installation/windows.html on both Windows 10 and Windows Server 2016. All tests produce the same exception.

```
PS C:\> cat .\test.txt
egg
bacon
spam
PS C:\> salt-call --local file.line test.txt sausage mode=insert after=bacon
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
NameError: global name '_assert_occurrence' is not defined
Traceback (most recent call last):
  File "C:\salt\bin\Scripts\salt-call", line 11, in <module>
    salt_call()
  File "C:\salt\bin\lib\site-packages\salt\scripts.py", line 391, in salt_call
    client.run()
  File "C:\salt\bin\lib\site-packages\salt\cli\call.py", line 58, in run
    caller.run()
  File "C:\salt\bin\lib\site-packages\salt\cli\caller.py", line 134, in run
    ret = self.call()
  File "C:\salt\bin\lib\site-packages\salt\cli\caller.py", line 204, in call
    ret['return'] = func(*args, **kwargs)
  File "C:\salt\bin\lib\site-packages\salt\modules\file.py", line 1778, in line
    _assert_occurrence(body, after, 'after')
NameError: global name '_assert_occurrence' is not defined
Traceback (most recent call last):
  File "C:\salt\bin\Scripts\salt-call", line 11, in <module>
    salt_call()
  File "C:\salt\bin\lib\site-packages\salt\scripts.py", line 391, in salt_call
    client.run()
  File "C:\salt\bin\lib\site-packages\salt\cli\call.py", line 58, in run
    caller.run()
  File "C:\salt\bin\lib\site-packages\salt\cli\caller.py", line 134, in run
    ret = self.call()
  File "C:\salt\bin\lib\site-packages\salt\cli\caller.py", line 204, in call
    ret['return'] = func(*args, **kwargs)
  File "C:\salt\bin\lib\site-packages\salt\modules\file.py", line 1778, in line
    _assert_occurrence(body, after, 'after')
NameError: global name '_assert_occurrence' is not defined
PS C:\>
```

The same test works fine on Linux.

```
root@ubuntu-xenial:~# cat test.txt 
egg
bacon
spam
root@ubuntu-xenial:~# salt-call --local file.line test.txt sausage mode=insert after=bacon
local:
    --- 
    +++ 
    @@ -1,3 +1,4 @@
     egg
     bacon
    +sausage
     spam
root@ubuntu-xenial:~# 
```

I'm quite mystified by this issue. The _assert_occurrence() function is clearly defined above line() in file.py on line 1568 (2017.7.2). It is accessible in the global scope on Linux, so why is Windows any different? 🙀

--- a/salt/modules/file.py
+++ b/salt/modules/file.py
@@ -1856,10 +1856,8 @@ def line(path, content, match=None, mode=None, location=None,
     if changed:
         if show_changes:
             with salt.utils.fopen(path, 'r') as fp_:
-                path_content = _splitlines_preserving_trailing_newline(
-                    fp_.read())
-            changes_diff = ''.join(difflib.unified_diff(
-                path_content, _splitlines_preserving_trailing_newline(body)))
+                path_content = fp_.read().splitlines(True)
+            changes_diff = ''.join(difflib.unified_diff(path_content, body.splitlines(True)))
         if __opts__['test'] is False:
             fh_ = None
             try:

--- a/salt/modules/win_file.py
+++ b/salt/modules/win_file.py
@@ -67,7 +67,7 @@ from salt.modules.file import (check_hash,  # pylint: disable=W0611
         lstat, path_exists_glob, write, pardir, join, HASHES, HASHES_REVMAP,
         comment, uncomment, _add_flags, comment_line, _regex_to_static,
         _get_line_indent, apply_template_on_contents, dirname, basename,
-        list_backups_dir)
+        list_backups_dir, _assert_occurrence, _starts_till)
 from salt.modules.file import normpath as normpath_
 
 from salt.utils import namespaced_function as _namespaced_function
@@ -98,7 +98,7 @@ def __virtual__():
             global write, pardir, join, _add_flags, apply_template_on_contents
             global path_exists_glob, comment, uncomment, _mkstemp_copy
             global _regex_to_static, _get_line_indent, dirname, basename
-            global list_backups_dir, normpath_
+            global list_backups_dir, normpath_, _assert_occurrence, _starts_till
 
             replace = _namespaced_function(replace, globals())
             search = _namespaced_function(search, globals())
@@ -165,6 +165,8 @@ def __virtual__():
             basename = _namespaced_function(basename, globals())
             list_backups_dir = _namespaced_function(list_backups_dir, globals())
             normpath_ = _namespaced_function(normpath_, globals())
+            _assert_occurrence = _namespaced_function(_assert_occurrence, globals())
+            _starts_till = _namespaced_function(_starts_till, globals())
 
             return __virtualname__
     return (False, "Module win_file: module only works on Windows systems")

--- a/tests/integration/modules/file.py
+++ b/tests/integration/modules/file.py
@@ -293,6 +293,16 @@ class FileModuleTest(integration.ModuleCase):
         else:
             self.assertItemsEqual(ret, ['file://' + self.myfile, 'filehash'])
 
+    def test_file_line_changes_format(self):
+        '''
+        Test file.line changes output formatting.
+
+        Issue #41474
+        '''
+        ret = self.minion_run('file.line', self.myfile, 'Goodbye',
+                              mode='insert', after='Hello')
+        self.assertIn('Hello\n+Goodbye', ret)
+
 if __name__ == '__main__':
     from integration import run_tests
     run_tests(FileModuleTest)

